pkg/scalers/azure/azure_aad_workload_identity.go

```diff
@@ -44,12 +44,16 @@ const (
 )
 
 // GetAzureADWorkloadIdentityToken returns the AADToken for resource
-func GetAzureADWorkloadIdentityToken(ctx context.Context, resource string) (AADToken, error) {
+func GetAzureADWorkloadIdentityToken(ctx context.Context, identityID, resource string) (AADToken, error) {
 	clientID := os.Getenv(azureClientIDEnv)
 	tenantID := os.Getenv(azureTenantIDEnv)
 	tokenFilePath := os.Getenv(azureFederatedTokenFileEnv)
 	authorityHost := os.Getenv(azureAuthrityHostEnv)
 
+	if identityID != "" {
+		clientID = identityID
+	}
```

Comment:
No, we should not. They could set a client id when installing KEDA. And use that client id across some some scalers such as Service Bus, or an Event Hub. But let's say they also want to access Key Vault, but they don't want to use the same identity as it would get too many permissions. They can use a TriggerAuthentication CRD and set the identity id which will be used.

Reply:
okey, thanks for the clarification 🙇 . Makes sense totally